epochStart/bootstrap/shardStorageHandler.go

```diff
@@ -524,6 +598,10 @@ func (ssh *shardStorageHandler) saveTriggerRegistry(components *ComponentsNeeded
 func getAllMiniBlocksWithDst(metaBlock *block.MetaBlock, destId uint32) map[string]block.MiniBlockHeader {
 	hashDst := make(map[string]block.MiniBlockHeader)
 	for i := 0; i < len(metaBlock.ShardInfo); i++ {
+		if metaBlock.ShardInfo[i].ShardID == destId {
+			continue
+		}
+
 		for _, val := range metaBlock.ShardInfo[i].ShardMiniBlockHeaders {
 			isCrossShardDestMe := val.ReceiverShardID == destId && val.SenderShardID != destId
```

Comment:
should the `isCrossShardDestMe := (val.ReceiverShardID == destId || **val.ReceiverShardID == core.AllShardId**) && val.SenderShardID != destId ` from line 616 be used also here?

Reply:
the name of the function here is a bit misleading

this intent of this method is to only returns the miniBlocks with destID as receiver shard, but only ones that are pending to destId shard due to their inclusion in this metablock header.

the miniBlocks with core.AllShardId as destination become pending for a shard only when included in the MetaBlock.MiniBlockHeaders, not when processed by other shards.

Reply:
ok